src/main/java/bookbob/functions/CommandHandler.java

```diff
@@ -413,6 +413,109 @@ public void edit(String input, Records records) throws IOException {
         FileHandler.autosave(records);
     }
 
+    public void editVisit(String input, Records records) throws IOException {
+        // Extract NRIC from input command
+        int nricStart = input.indexOf("ic/");
+        assert nricStart != -1 : "Please provide a valid patient NRIC in the records.";
+        if (nricStart == -1) {
+            System.out.println("Please provide a valid patient NRIC.");
+            return;
+        }
+        int nricEnd = findNextFieldStart(input, nricStart + 3);
+        String nric = input.substring(nricStart + 3, nricEnd).trim();
+
+        Patient patient = null;
+
+        // Search for the patient by NRIC
+        for (Patient p : records.getPatients()) {
+            if (p.getNric().trim().replaceAll("\\s+", "")
+                    .equalsIgnoreCase(nric.replaceAll("\\s+", "").trim())) {
+                patient = p;
+                break;
+            }
+        }
+
+        if (patient == null) {
+            System.out.println("No patient found with the given NRIC.");
+            return;
+        }
+
+        // Extract visit date from input command
+        int dateStart = input.indexOf("date/");
+        assert dateStart != -1 : "Please provide a valid visit date.";
+        if (dateStart == -1) {
+            System.out.println("Please provide a valid visit date.");
+            return;
+        }
+        int dateEnd = findNextFieldStart(input, dateStart + 5);
```

Comment:
A comment in general that could be discussed with the team: 

We could replace these "magic numbers" with a constant instead?

Reply:
Great idea!